### DISCUSSION TITLE
mirai-only regression test oddities

### DISCUSSION THREAD

**Discussion by wlandau** [wlandau/crew#126] [Q&A] [answered]
I am using https://github.com/shikokuchuo/mirai/commit/ca2dcedff0d0b809d57ad1262a8b8c7f3f110dd8 and https://github.com/shikokuchuo/nanonext/commit/7821bcd1aa0bc3c7cf97884d26819f478efbcc39, and I am revisiting the regression test at https://github.com/wlandau/crew/blob/main/tests/mirai/test-tls-max_tasks.R. When I run it locally on my MacBook, there are many pushed tasks that do not complete, and my code does not seem to know how to launch new worker instances to run them. To launch new worker instances, this test looks for workers that connected and then exited:

https://github.com/wlandau/crew/blob/91e1ea271eb1a20e294f412e45735546d95bca2c/tests/mirai/test-tls-max_tasks.R#L34-L37

I figured out that `info[, "instance"] > 0L` was not happening because `instance` ended up as `-1` for  all workers. @shikokuchuo, should this be possible?

```r
mirai::status()$daemons
#>                                                   i online instance assigned complete
#> wss://127.0.0.1:5000/1/5569f0daa8a94437965d3a96   1      0       -1      100      100
#> wss://127.0.0.1:5000/2/6c8dea659d2b698d35de57cc   2      0       -1      100      100
#> wss://127.0.0.1:5000/3/c2c576c365d6cdfb1ffcd500   3      0       -1      100      100
#> wss://127.0.0.1:5000/4/f51b02e0401d52bf2d155bb8   4      0       -1      100      100
#> wss://127.0.0.1:5000/5/328b6d66eac7d1770957e39f   5      0       -1      100      100
#> wss://127.0.0.1:5000/6/4670db6dc719fd29b7fc8e09   6      0       -1      100      100
#> wss://127.0.0.1:5000/7/7895eeda095b17f833e3d051   7      0       -1      100      100
#> wss://127.0.0.1:5000/8/66395ca7d6c4dc8476675360   8      0       -1      100      100
#> wss://127.0.0.1:5000/9/0ab84f89178b2e0b3d3cada1   9      0       -1      100      100
#> wss://127.0.0.1:5000/10/5815f32bdcce33c9ef2b1967 10      0       -1      100      100
#> wss://127.0.0.1:5000/11/1ca7cc6034417fcb9262eb4e 11      0       -1      100      100
#> wss://127.0.0.1:5000/12/4035e2f8d12ee717cb9c0975 12      0       -1      100      100
#> wss://127.0.0.1:5000/13/f5fd4e4edac06493bcb83ffb 13      0       -1      100      100
#> wss://127.0.0.1:5000/14/43e994277dc3beda66e0718c 14      0       -1      100      100
#> wss://127.0.0.1:5000/15/6ad89ef82272980586355888 15      0       -1      100      100
#> wss://127.0.0.1:5000/16/a96b217efa25d1ea071b97f4 16      0       -1      100      100
#> wss://127.0.0.1:5000/17/e5623949eb568a59ce7662d9 17      0       -1      100      100
#> wss://127.0.0.1:5000/18/f902a2299ef0f4c460be059f 18      0       -1      100      100
#> wss://127.0.0.1:5000/19/99087d92ac20d6cc270d57e7 19      0       -1      100      100
#> wss://127.0.0.1:5000/20/21a6dfaf26901b34ac19de6e 20      0       -1      100      100
```

**Comment by wlandau** [comment]
I would expect `instance` to be > 1L at the time I called `status()` because no workers were running (all had exited on their own).

**Reply by shikokuchuo** [reply, marked as answer]
-1 simply means that `saisei()` has been called after the first launch. And then subsequently no daemons have managed to connect.

Not surprising as they were all trying to connect at the same original URL. Fixed in #127.

If these no longer follow the logic in `crew` it probably makes sense to put these elsewhere, as it seems they are no longer serving a purpose.

**Reply by wlandau** [reply]
Ah of course, I forgot to update the socket on my end! Thanks so much for catching that! And it's just staggering how much faster this test is running compared to the current CRAN mirai!

I do prefer to keep these tests because they might catch potential regressions later on.
